src/DriverManager.php

```diff
@@ -299,6 +300,10 @@ private static function parseDatabaseUrl(array $params): array
 
         // (pdo_)?sqlite3?:///... => (pdo_)?sqlite3?://localhost/... or else the URL will be invalid
         $url = preg_replace('#^((?:pdo_)?sqlite3?):///#', '$1://localhost/', $params['url']);
+        if ($url === null) {
+            throw new LogicException(preg_last_error_msg(), preg_last_error());
+        }
```

Comment:
I'm confident that this case never happens because the regular expressions are analyzed by PHPStan already. But apparently, it is desired that we perform an explicit `null` check here. See https://github.com/phpstan/phpstan/issues/1901#issuecomment-463973500 for a discussion on this topic.

Reply:
Even when I manage to get a `Process exited with code 137.`, there will be a string as a return type. I don't know if an evil regex could end up in a return null after some runtime, but I don't see one in our use case. I guess we can live with a null check, but if you're sure about not-null, then maybe an `assert` is sufficient?

Reply:
Agreed, let's go with an assert.